lib/util/memcache.ex

```diff
@@ -73,6 +73,31 @@ defmodule Antikythera.Memcache do
     end)
   end
 
+  @doc """
+  Try to read a value associated with the `key` from #{inspect(__MODULE__)}.
+
+  `fun` is evaluated if a value is not found, and the new value returned by `fun` is stored in #{inspect(__MODULE__)}.
+  If a value is found in #{inspect(__MODULE__)} or writing the new value to #{inspect(__MODULE__)} succeeds, the value is returned as `{:ok, value}`, but if writing the new value fails, an error is returned in the same manner as `write/5`.
+
+  Parameters `lifetime_in_sec` and `prob_lifetime_ratio` are used to call `write/5` and the details are described above.
+  """
+  defun read(key                 :: Key.t,
```

Comment:
Concerning the function name:
I'd like to somehow express that this function is side-effecting one.
`read_or_write` looks as if "read" and "write" are on an equal footing (which they are not).
How about something like `read_or_else_write_with_fun` (a bit lengthy but hopefully easier to read)?
Any thoughts?

Reply:
Indeed, the name is confusing.
`read_or_else_write_with_fun` is well-understandable, but I think `read_or_else_write` is sufficient to express the meaning.

Reply:
OK, let's go with `read_or_else_write`!

Reply:
Renamed in 2e944a195874f417aa0cb8845c1a6a39582f894f